### ISSUE TITLE
User Groups - Array of Values

### ISSUE THREAD

**Issue by masonry47** [flipboxfactory/saml-sp#202]
Hi there!  

We are attempting to assign users into groups and are using the config 'groupAttributeNames' to match the name of the attribute. So our config file looks like this

```
return [
   'groupAttributeNames' => [ 'roles', ],
];
```

And the attribute we are using is this (always returning multiple values). 

```
            <saml:Attribute Name="roles" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
                <saml:AttributeValue xsi:type="xs:string">basic_access</saml:AttributeValue>
                <saml:AttributeValue xsi:type="xs:string">standard_access</saml:AttributeValue>
            </saml:Attribute>
```
So the issue is users are successfully being assigned but only to the the first value, where we would expect them to be assigned to both user groups.  We did ensure the user group handles match the attribute values. 

Thanks in advance for any thoughts on this!

**Comment by dsmrt** [comment]
I haven't seen this behavior but I'll spin things up and give it a test.

Just to be sure, both of those users groups are created in your Craft instance right?

**Comment by masonry47** [comment]
We actually figured it out. We think because of this line 

https://github.com/flipboxfactory/saml-sp/blob/c1991f64aa77815f3671fd4435fca3d2d884ce93/src/services/login/UserGroups.php#L39C20-L39C20

Group handles needed to be camel case.  We were matching the attribute value. So for example for this value, "basic_access", we were making the group handle in Craft match that exactly. If we changed the handle to "basicAccess" all works as expected.  

Thanks!
